2_process/src/process_nhdv2_attr.R

```diff
@@ -269,3 +269,184 @@ process_catchment_nhdv2_attr <- function(file_path,vars_table,segs_w_comids,nhd_
+  #' @param prms_reach_attr the PRMS reach attribute tbl 
+  #' @param prms_nhdv2_xwalk the crosswalk tbl from NHD reaches to PRMS reaches
+  #' @param nhdv2_reaches the NHD reaches as an sf object
+  #' @param prms_attribute_df PRMS attribute table with all_from_segs discovered 
+  #' from recursive function (output of p2_prms_attribute_df target)
+  #' @param drop_columns character vector of column names to remove from nhdv2_attr
+  #' 
+  #' @value Returns a refined nhdv2_attr based on the columns to drop
+  
+  #Detect variables that are all equal across the modeling domain and remove them
+  #removes "BEDPERM_4" and "HGAC"
+  unique_col_vals <- apply(nhdv2_attr, 2, FUN = function(x) length(unique(x)))
+  nhdv2_attr_refined <- nhdv2_attr[, which(unique_col_vals > 1)] %>%
+    #Remove other columns
+    #PHYSIO_AREA says which proportion of catchments are covered by physiographic regions
+    #RUN7100 seems like it is by HUC02 instead of reach.
+    select(!contains(drop_columns)) %>%
+    #Modify the CAT Basin Areas that are 0 with PRMS areas
+    #These areas are otherwise nearly identical (max difference of 0.1 sq.km)
+    mutate(CAT_BASIN_AREA_sum = case_when(CAT_BASIN_AREA_sum == 0 ~ AREASQKM_PRMS,
+                                          TRUE ~ CAT_BASIN_AREA_sum)) %>%
+    #drop PRMS area column
+    select(-AREASQKM_PRMS)
+  
+  #RECHG
+  #Change recharge for NA segment to the average of its neighbors (from_segs and to_seg)
+  #index to change
+  nhdv2_attr_refined$CAT_RECHG_area_wtd <- refine_from_neighbors(nhdv2_attr_refined,
+                                                                 attr_i = 'CAT_RECHG_area_wtd', 
+                                                                 prms_reach_attr)
+  
+  #EWT - water table
+  #Change EWT for segment with deep value to the average of its neighbors (from_segs and to_seg)
+  nhdv2_attr_refined$CAT_EWT_area_wtd <- refine_from_neighbors(nhdv2_attr_refined,
+                                                               attr_i = 'CAT_EWT_area_wtd',
+                                                               prms_reach_attr)
+  
+  #STRM_DENS
+  #Compute stream density from the NHD catchment reach length and area
+  #only for the 5 NA PRMS segments. These have 1 or 2 NHD catchments.
+  # other PRMS segments with some NA stream densities cover areas <3% of total area.
+  #Gather the PRMS areas for these reaches
+  ind_areas <- filter(nhdv2_attr_refined, is.na(CAT_STRM_DENS_area_wtd)) %>%
+    select(PRMS_segid, CAT_BASIN_AREA_sum)
+  #Gather the sum of NHD reach lengths in m
+  ind_areas$length_m <- 0
+  for (i in 1:nrow(ind_areas)){
+    #all NHD reaches for this PRMS segment
+    nhd_reaches <- filter(prms_nhdv2_xwalk,
+                          PRMS_segid %in% ind_areas$PRMS_segid[i]) %>%
+      select(comid_cat) %>%
+      str_split(., pattern = ';', simplify = T)
+    
+    ind_areas$length_m[i] <- filter(nhdv2_reaches,
```

Comment:
`ind_areas$length_m` indicates meters but the NHDPlus attributes for length are given in units of KM (LENGTHKM). I see the comment below about the magnitude of the stream density estimates, but I think that is just because the basin areas are _very_ small and not because of incorrect units on the length attributes:

Given your comment I quickly checked a couple things. 1) the length values using `sf::st_length` (shown below as check_length_m):
```
> ind_areas
# A tibble: 5 x 4
  PRMS_segid CAT_BASIN_AREA_sum check_length_m length_m
  <chr>                   <dbl>          <dbl>    <dbl>
1 282_1                 0.0009          0.0634    0.064
2 287_1                 0.00230         0.0480    0.048
3 878_1                 0.00365         0.0620    0.062
4 892_1                 0.00470         0.0471    0.047
5 894_1                 0.0018          0.0526    0.053
```

2) how the calculated values match up with the values from `CAT_STRM_DENS_area_wtd`
![trdens](https://user-images.githubusercontent.com/8785034/159812170-743d733c-a464-41d5-b47d-e5e1c9269051.png)

From these checks, I think the calculation/units are correct but these catchments are wonky.

Reply:
Yes, I also think that the length units are in km. The ScienceBase attribute description for stream density is the length in m / area in km^2, and it seems instead that it's length in km / area in km^2. I changed the variable name to `length_km` and edited the comment